include/bsp/iotc_bsp_io_fs.h

```diff
@@ -32,223 +29,175 @@
+  /** Open and write to the file. */
   IOTC_BSP_IO_FS_OPEN_WRITE = 1 << 1,
-  /** Open the file to append. **/
+  /** Open and append to the file. */
   IOTC_BSP_IO_FS_OPEN_APPEND = 1 << 2,
 } iotc_bsp_io_fs_open_flags_t;
 
 /**
- * @function
- * @brief Determine file size.
- *
- * @param [in] resource_name File name.
- * @param [out] resource_stat A structure for the file size information.
+ * @brief Gets the size of a file.
  *
- * @see iotc_bsp_io_fs_state_e
- *
- * @return iotc_bsp_io_fs_state_t File management function status.
+ * @param [in] resource_name The file name.
+ * @param [out] resource_stat The size, in bytes, of the file.
  */
 iotc_bsp_io_fs_state_t iotc_bsp_io_fs_stat(
     const char* const resource_name, iotc_bsp_io_fs_stat_t* resource_stat);
 
 /**
- * @function
- * @brief Open a file.
- *
- * Implementations of this function must return a
- * iotc_bsp_io_fs_resource_handle_t handle that's passed to subsequent
- * operations (read, write, or close). The Device SDK calls this function and
- * then passes resource_handle_out to <code>iotc_bsp_io_fs_read()</code> or
- * <code>iotc_bsp_io_fs_write()</code> to perform the corresponding operation.
- *
- * @param [in] resource_name File name.
- * @param [in] size (Optional) The size, in bytes, of the file. Required on
- *     non-POSIX implementations.
- * @param [in] open_flags An iotc_bsp_io_fs_open_flags_t bitmask.
- * @param [out] resource_handle_out A pointer to an
- *     iotc_bsp_io_fs_resource_handle handle.
- *
- * @see iotc_bsp_io_fs_read
- * @see iotc_bsp_io_fs_write
- * @see iotc_bsp_io_fs_state_e
- * @see iotc_bsp_io_fs_open_flags_t
- * @see iotc_bsp_io_fs_resource_handle_t
- *
- * @return iotc_bsp_io_fs_state_t File management function status.
+ * @details Opens a file.
+ *
+ * @param [in] resource_name The filename.
+ * @param [in] size The size, in bytes, of the file.
+ * @param [in] open_flags A {@link ::iotc_bsp_io_fs_open_flags_t file operation}
+ *     bitmask.
+ * @param [out] resource_handle_out A
+ *     {@link ::iotc_bsp_io_fs_resource_handle_t handle to an open file}.
  */
 iotc_bsp_io_fs_state_t iotc_bsp_io_fs_open(
     const char* const resource_name, const size_t size,
     const iotc_bsp_io_fs_open_flags_t open_flags,
     iotc_bsp_io_fs_resource_handle_t* resource_handle_out);
 
 /**
- * @function
- * @brief Read a file.
- *
- * The Device SDK passes an iotc_bsp_io_fs_resource_handle_t handle to
- * this function in order to read a file. Before calling this function,
- * the Device SDK calls the <code>iotc_bsp_io_fs_open()</code> function
- * to create an iotc_bsp_io_fs_resource_handle_t handle.
+ * @brief Reads a file.
  *
- * Implementations of this function must fill the buffer parameter at
- * offset 0.
+ * @details The function must fill the buffer at offset 0. The function can
+ * allocate buffers by:
+ *     - Allocating the buffer once, resusing it at each function call,
+ *       and freeing it when the file closes.
```

Comment:
This isn't trivial to understand from this description, add a comment: See the POSIX impl bsp/platform/posix/iotc_bsp_io_fs_posix.c.

Reply:
Done. Do you think any of the content in the previous/original draft would help, or is this just something best explained with an example?

Reply:
I think it is good now with example. If one implements it, then he is a developer, example code is the most accurate description.